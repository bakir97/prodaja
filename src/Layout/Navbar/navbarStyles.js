export default {
  root: {
    flexGrow: 1
  },
  grow: {
    flexGrow: 1
  },
  navBar: {
    background: "linear-gradient(to right, #373b44, #4286f4)"
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  }
};
