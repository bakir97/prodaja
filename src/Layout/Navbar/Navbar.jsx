import React, { Component } from "react";
import { connect } from "react-redux";
import { logout } from "../../redux/actions/loginAction";
// material styles
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
// end material styles
import Drawer from "./Drawer/Drawer";
import navbarStyles from "./navbarStyles";
import Logout from "../../assets/logout.png";

const styles = navbarStyles;
class Navbar extends Component {
  state = {
    open: false
  };
  promjenaDrawera = () => {
    this.setState(prevState => ({
      open: !prevState.open
    }));
  };
  handleLogout = () => {
    this.props.logout();
    this.props.history.push("/");
  };
  render() {
    const { auth } = this.props;
    const { root, menuButton, navBar, grow } = this.props.classes;
    const { open } = this.state;
    return (
      <div className={root}>
        <AppBar className={navBar} position="fixed">
          <Toolbar>
            <IconButton
              onClick={this.promjenaDrawera}
              className={menuButton}
              color="inherit"
              aria-label="Menu"
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" color="inherit" className={grow}>
              Sell It
            </Typography>
            {Object.keys(auth).length > 0 && (
              <>
                <Typography
                  variant="h6"
                  color="inherit"
                  style={{ marginRight: 5 }}
                >
                  {auth.username}
                </Typography>
                <IconButton onClick={this.handleLogout}>
                  {<img src={Logout} style={{ paddingLeft: 5 }} alt="" />}
                </IconButton>
              </>
            )}
          </Toolbar>
        </AppBar>
        <Drawer
          auth={auth}
          open={open}
          promjenaDrawera={this.promjenaDrawera}
        />
      </div>
    );
  }
}
const mapStateToProps = state => ({
  auth: state.login.account
});

const mapDispatchToProps = { logout };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(Navbar));
