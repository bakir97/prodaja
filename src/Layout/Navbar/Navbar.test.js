import { Navbar } from "./Navbar";
describe("Navbar", () => {
  let admin = true;
  const wrapper = shallow(<Navbar auth={admin} classes={{}} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
