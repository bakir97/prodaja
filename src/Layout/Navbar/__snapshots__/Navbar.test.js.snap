// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<div>
  <WithStyles(AppBar)
    position="fixed"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        color="inherit"
        onClick={[Function]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        variant="h6"
      >
        Sell It
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(SwipeableTemporaryDrawer)
    auth={true}
    open={false}
    promjenaDrawera={[Function]}
  />
</div>
`;
